### PR TITLE
feat(sol): Support versioned transaction signing

### PR DESCRIPTION
**Pull request by Dorac**
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-20066 Support SOL versioned transaction](https://app.clickup.com/t/25577573/CW-20066)

File: packages/coin-sol/src/index.ts [hunk 1]
```diff
@@ -219,6 +219,25 @@ class Solana extends COIN.EDDSACoin implements COIN.Coin {
     return sign.signMessage(signMsgData, script, argument);
   }
 
+  async signAllTransaction(signTxData: types.signTransactionType[]): Promise<string> {
```
Comment (gundomlegend) [inline packages/coin-sol/src/index.ts:222]: type 應該是有兩種，有可能是 Transaction[] 或是 VersionedTransaction[]

Comment (Dorac) [reply packages/coin-sol/src/index.ts]: Remove signAllTransaction temporarily, to be implemented after validating signTransaction.

File: packages/coin-sol/src/message/legacy.ts [hunk 2]
```diff
@@ -477,6 +480,70 @@ class Message {
 
     return signData.slice(0, length + instructionBuffer.length).toString('hex');
   }
+
+  /**
+   * Decode a compiled message into a Message object.
+   */
+  static from(buffer: Buffer | Uint8Array | Array<number>): Message {
```
Comment (gundomlegend) [inline packages/coin-sol/src/message/legacy.ts:487]: 這邊什麼時候會用到 Message.from? MessageV0 似乎沒看到 from method

File: packages/coin-sol/src/message/versioned.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    // the lower 7 bits of the prefix indicate the message version
+    return maskedPrefix;
+  },
+
+  deserialize: (serializedMessageHex: string): VersionedMessageType => {
+    const serializedMessage = Buffer.from(serializedMessageHex, 'hex');
+    const version =
+      VersionedMessage.deserializeMessageVersion(serializedMessage);
+    if (version === 'legacy') {
+      return Message.from(serializedMessage);
```
Comment (gundomlegend) [inline packages/coin-sol/src/message/versioned.ts:26]: 看到有在這邊使用，想說可以統一用 deserialize 或是 from XD

**Review by gundomlegend — APPROVED** [review]

**Review by hankliu418 — APPROVED** [review]

**Pull request merged**